app/models/group_category.rb

```diff
@@ -0,0 +1,18 @@
+class GroupCategory < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :category_items, dependent: :destroy
+  has_many :entity_categories, through: :category_items
+
+  def total_amount
+    entity_categories.sum(:amount)
+  end
+
+  ICONS = {
+    'fa-house' => 'Home',
+    'fa-cart-shopping' => 'Shopping Cart',
+    'fa-car' => 'Car',
+    'fa-truck-medical' => 'Medical Truck',
+    'fa-piggy-bank' => 'Piggy Bank',
+    'fa-credit-card' => 'Credit Card'
+  }.freeze
```

Comment:
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#:~:text=You%20should%20validate%20all%20user%20input%20to%20make%20sure%20that%20anyone%20with%20bad%20intentions%20cannot%20compromise%20your%20app.), Please add validations to validate all user input (`:name` and `:icon`) to make sure that anyone with bad intentions cannot compromise your app.